crates/extrinsics/src/env_check.rs

```diff
@@ -0,0 +1,371 @@
+use ink_metadata::InkProject;
+use scale_info::{
+    form::PortableForm,
+    Field,
+    PortableRegistry,
+    TypeDef,
+};
+
+use anyhow::{
+    Context,
+    Result,
+};
+
+fn get_node_env_fields(registry: &PortableRegistry) -> Result<Vec<Field<PortableForm>>> {
+    let env_type = registry
+        .types
+        .iter()
+        .find(|t| t.ty.path.segments.contains(&"Environment".to_string()))
```

Comment:
I had an idea to make this more accurate, and use the full namespace here and have it work in the tests, we could use the recently added feature: https://github.com/paritytech/scale-info/pull/197

Reply:
Good one